draft-ietf-quic-http.md

```diff
@@ -181,93 +181,70 @@ application.
+and is analogous to HTTP/2's Stream 0.  This control stream is considered
+critical to the HTTP connection.  If the control stream is closed for any
+reason, this MUST be treated as a connection error of type
 QUIC_CLOSED_CRITICAL_STREAM.
 
 When HTTP headers and data are sent over QUIC, the QUIC layer handles most of
-the stream management. An HTTP request/response consumes a pair of streams: This
-means that the client's first request occurs on QUIC streams 3 and 5, the second
-on stream 7 and 9, and so on. The server's first push consumes streams 2 and 4.
-This amounts to the second least-significant bit differentiating the two streams
-in a request.
-
-The lower-numbered stream is called the message control stream and carries
-frames related to the request/response, including HEADERS. The higher-numbered
-stream is the data stream and carries the request/response body with no
-additional framing. Note that a request or response without a body will cause
-this stream to be half-closed in the corresponding direction without
-transferring data.
-
-Because the message control stream contains HPACK data which manipulates
-connection-level state, the message control stream MUST NOT be closed with a
-stream-level error.  If an implementation chooses to reject a request with a
-QUIC error code, it MUST trigger a QUIC RST_STREAM on the data stream only.  An
-implementation MAY close (FIN) a message control stream without completing a
-full HTTP message if the data stream has been abruptly closed.  Data on message
-control streams MUST be fully consumed, or the connection terminated.
-
-All message control streams are considered critical to the HTTP connection.  If
-a message control stream is terminated abruptly for any reason, this MUST be
-treated as a connection error of type HTTP_RST_CONTROL_STREAM.  When a message
-control stream terminates cleanly, if the last frame on the stream was
-truncated, this MUST be treated as a connection error (see HTTP_MALFORMED_* in
-{{http-error-codes}}).
-
-Pairs of streams must be utilized sequentially, with no gaps.  The data stream
-is opened at the same time as the message control stream is opened and is closed
-after transferring the body.  The data stream is closed immediately after
-sending the request headers if there is no body.
+the stream management. An HTTP request/response consumes a single stream: This
+means that the client's first request occurs on QUIC stream 3, the second on
+stream 5, and so on. The server's first push consumes stream 2.
+
+This stream carries frames related to the request/response (see {{frames}}).
+When a stream terminates cleanly, if the last frame on the stream was truncated,
+this MUST be treated as a connection error (see HTTP_MALFORMED_* in
+{{http-error-codes}}).  Streams which terminate abruptly may do so at any point
+in the frame.
+
+Streams used for requests MUST be utilized sequentially, with no gaps.  Streams
```

Comment:
is this a sender side only restriction, because there doesn't seem to be a way to enforce this on the receiver